### ISSUE TITLE
GitLab Logger generates invalid links

### ISSUE THREAD

**Issue by vincentchalamon** [infection/infection#1977]
| Question    | Answer
| ------------| ---------------
| Infection version | >=0.28.1
| Test Framework version | PHPUnit 11.1
| PHP version | 8.3.7
| Platform    | Docker Compose project with a `php:8.3-fpm` (Debian) container
| GitHub Repo | N/A
| GitLab Repo | (private)

The GitLab Logger generates invalid links on infection report. Using infection/infection:0.27.11 and 0.28.0, the report is fine:

![image](https://github.com/infection/infection/assets/407859/0e136e7e-47fb-4752-955f-c318bb29d9ec)

Upgrading to infection/infection:0.28.1 or higher generates invalid links:

![image](https://github.com/infection/infection/assets/407859/3a72e482-a2be-411c-90cf-c5411f290b13)

This may have been introduced here: https://github.com/infection/infection/commit/1ee4ffa1f971e74c285d2cfc344cead82055df6e

<details>
 <summary>.gitlab-ci.yml</summary>

  ```yaml
variables:
    # Use TLS https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#tls-enabled
    DOCKER_HOST: tcp://docker:2376
    DOCKER_TLS_CERTDIR: "/certs"
    DOCKER_DRIVER: overlay2
    DOCKER_BUILDKIT: 1
    CONTAINER_IMAGE: $CI_REGISTRY_IMAGE:$CI_COMMIT_REF_SLUG
    # https://docs.docker.com/compose/environment-variables/envvars/#compose_file
    COMPOSE_FILE: compose.yaml:compose.override.ci.yaml
    WORKDIR: /shared/httpd/ae-api

stages:
    - build
    - test

build:
    stage: build
    image: docker:latest
    services:
        - docker:dind
    before_script:
        - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
    script:
        - docker buildx bake --push -f compose.yaml -f compose.override.ci.yaml php

phpunit-unit:
    stage: test
    needs:
        - build
    image: $CONTAINER_IMAGE
    variables:
        XDEBUG_MODE: coverage
    script:
        - cd $WORKDIR
        - bin/phpunit -c phpunit.xml.dist --testsuite=Unit --coverage-xml=$CI_PROJECT_DIR/build/coverage/coverage-xml
    artifacts:
        when: on_success
        expire_in: 1 week
        paths:
            # required by "infection" job
            - build/coverage/

infection:
    stage: test
    needs:
        - phpunit-unit
    dependencies:
        - phpunit-unit
    image: $CONTAINER_IMAGE
    allow_failure: true
    script:
        - cd $WORKDIR
        - vendor/bin/infection --threads=4 --skip-initial-tests --only-covered --coverage=$CI_PROJECT_DIR/build/coverage --logger-gitlab=$CI_PROJECT_DIR/infection-gitlab.json
    artifacts:
        when: on_success
        expire_in: 1 week
        reports:
            codequality: infection-gitlab.json
 ```
</details>

<details>
 <summary>phpunit.xml</summary>

 ```xml
<?xml version="1.0" encoding="UTF-8"?>

<!-- https://phpunit.readthedocs.io/en/latest/configuration.html -->
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
         backupGlobals="false"
         colors="true"
         bootstrap="tests/phpunit/bootstrap.php"
         failOnDeprecation="true"
>
    <php>
        <ini name="display_errors" value="1" />
        <ini name="error_reporting" value="-1" />
        <env name="APP_ENV" value="test" force="true" />
        <env name="SYMFONY_DEPRECATIONS_HELPER" value="max[direct]=0" />
        <env name="LANG" value="en_GB.UTF-8" force="true" />
        <server name="APP_ENV" value="test" force="true" />
        <server name="SHELL_VERBOSITY" value="-1" />
        <server name="SYMFONY_PHPUNIT_REMOVE" value="" />
        <server name="SYMFONY_PHPUNIT_VERSION" value="9.5" />
        <server name="KERNEL_CLASS" value="AE\Common\Infrastructure\Symfony\Kernel"/>
    </php>

    <source>
        <include>
            <directory>src</directory>
        </include>
    </source>

    <testsuites>
        <testsuite name="Unit">
            <directory>tests/phpunit/Unit</directory>
        </testsuite>
    </testsuites>

    <coverage
            ignoreDeprecatedCodeUnits="false"
            includeUncoveredFiles="true">
    </coverage>
</phpunit>
 ```
</details>

**Comment by maks-rafalko** [comment]
Thanks for such a detailed bug report. If you can help with fixing it, that would be really cool to speed it up.

**Comment by vincentchalamon** [comment]
Of course I'll investigate further and open a PR if I find a solution. Keep in touch!

**Comment by vincentchalamon** [comment]
Fix available at #1978, ready for review.

Sorry for the GitHub Actions queue, a related fix has been proposed in #1979.
